base-system/usrroot/usr/lib/hsync/libhapply.so

```diff
@@ -404,3 +406,23 @@ set_wallpaper() {
 	log "-Couldn't find any background to restore"
 	return 1 # error
 }
+
+set_scheduled_shutdown() {
+	local SHUTDOWN_TIME SHUTDOWN_TIME_EPOCH CURRENT_TIME_EPOCH SECONDS_TO_SHUTDOWN MINUTES_TO_SHUTDOWN
+	SHUTDOWN_TIME="$(get_directives_var Global ShutdownTime)"
+	log "Cancelling scheduled shutdown (if any)"
+	shutdown -c
+	if [ "$SHUTDOWN_TIME" = "none" ]; then
+		return 0
+	fi
+	SHUTDOWN_TIME_EPOCH=$(date -d "$SHUTDOWN_TIME" +%s)
+	CURRENT_TIME_EPOCH=$(date +%s)
+	SECONDS_TO_SHUTDOWN=$((SHUTDOWN_TIME_EPOCH-CURRENT_TIME_EPOCH))
+	if [ "$SECONDS_TO_SHUTDOWN" -lt 0 ]; then
+		log "Current shutdown is for a time that already happened, skiping scheduled shutdown."
+		exit 0
+	fi
+	MINUTES_TO_SHUTDOWN=$((SECONDS_TO_SHUTDOWN/60))
```

Comment:
Let's do a rounding here. For a minute, if less than 30s then 0min (floor), if more than 30s then 1min (ceil). E.g. 1:45 then the system will shutdown in 2min. 

This is because I noticed the following: Scheduled a system mode change on 00:05 min, and shutdown at 00:06 min. Then on the 00:05 sync, the system directly shutdown without firstly changing the mode, this was because at the time of the check, the seconds to shutdown were 54s, which leaded to shutdown now.

Reply:
Okay, added a rounding.